### ISSUE TITLE
WS001 check fail

### ISSUE THREAD

**Issue by jimmysung-devops** [docToolchain/asciidoc-linter#23]
use https://github.com/docToolchain/asciidoc-linter/blob/b364434dcf5ba7ff1ef709072d9e747c9676c776/docs/manual/rules.adoc provide vaild format, but check fail with WS001.
= Document Title (Level 1)

== Section (Level 2)

=== Subsection (Level 3)

== Another Section (Level 2)

Install tool: 
# Clone the repository
git clone https://github.com/docToolchain/asciidoc-linter.git
(get revision is b364434dcf5ba7ff1ef709072d9e747c9676c776 on main branch)

# Navigate to the project directory
cd asciidoc-linter

# Install the package
pip install .

**Comment by stuartpurgavie** [comment]
Also getting false positives on WS001. However, the output doesn't explicitly say WS001, which is another potential bugfix.

Passes:

```asciidoc
// Test comment
:adr: Architecture Decision Record

== {adr}: Section title test

[%hardbreaks]
text
more text

```

Fails:

```asciidoc
// Test comment
:adr: Architecture Decision Record

== {adr}: Section title test

[%hardbreaks]
text
more text

=== Another Header

Some paragraph
text

```

The fail case error messages aren't particularly useful:

```text
Results for ./000-TestLinter.adoc:
✗ ./000-TestLinter.adoc, line 1: Section title should be followed by a blank line
✗ ./000-TestLinter.adoc, line 2: Section title should be preceded by a blank line

```

There's not even a section title on lines 1 or 2, so something is definitely weird with the parser.
